### PR TITLE
Prevent ingesting samples older than `past_grace_period`

### PR DESCRIPTION
**Pull request by colega**
#### What this PR does

This PR implements a new limit, similar to `creation_grace_period`, that limits the age of the ingested samples based on the wall clock.

It also takes into account the OOO window, to avoid having to change both settings when OOO window is modified.

The default value of 1 week should cover any observability needs. If some customer wants to backfill their samples in order, they can increase the value of that to `100y` in the config to cover 100 years, for example.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/ingester/lifecycle_test.go [hunk 1]
```diff
@@ -20,14 +20,18 @@ import (
...
 	"github.com/stretchr/testify/require"
 
 	"github.com/grafana/mimir/pkg/ingester/client"
 	"github.com/grafana/mimir/pkg/util/validation"
 )
 
-const userID = "1"
+const (
+	userID              = "1"
+	testPastGracePeriod = 100 * 365 * 24 * time.Hour // 100 years, as many tests push samples around unix time 0 and we don't want to keep fixing this tests again and again.
```
Comment (pstibrany) [inline pkg/ingester/lifecycle_test.go:33]: We can just disable it, right?

File: pkg/util/validation/limits.go [hunk 2]
```diff
@@ -251,7 +253,9 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
...
 	_ = l.CreationGracePeriod.Set("10m")
-	f.Var(&l.CreationGracePeriod, CreationGracePeriodFlag, "Controls how far into the future incoming samples and exemplars are accepted compared to the wall clock. Any sample or exemplar will be rejected if its timestamp is greater than '(now + grace_period)'. This configuration is enforced in the distributor and ingester.")
+	f.Var(&l.CreationGracePeriod, CreationGracePeriodFlag, "Controls how far into the future incoming samples and exemplars are accepted compared to the wall clock. Any sample or exemplar will be rejected if its timestamp is greater than '(now + creation_grace_period)'. This configuration is enforced in the distributor and ingester.")
+	_ = l.PastGracePeriod.Set("1w")
+	f.Var(&l.PastGracePeriod, PastGracePeriodFlag, "Controls how far into the past incoming samples and exemplars are accepted compared to the wall clock. Any sample or exemplar will be rejected if its timestamp is lower than '(now - ooo window - past_grace_period)'. This configuration is enforced in the distributor and ingester.")
```
Comment (pstibrany) [inline pkg/util/validation/limits.go:258]: super-nit: `ooo` -> `OOO`?

File: pkg/ingester/lifecycle_test.go (hunk 1 above)
Comment (colega) [reply pkg/ingester/lifecycle_test.go:33]: I thought about using `0` value as `disabled`, but something inside of me didn't like it. I think it's fair to expect some kind of config set up.

Comment (pstibrany) [reply pkg/ingester/lifecycle_test.go:33]: Yeah, that's what I was suggesting. We can set reasonable default (once it's tested), but having ability to disable would be nice.

Comment (colega) [reply pkg/ingester/lifecycle_test.go:33]: So you're suggesting to use zero value to disable it?

File: pkg/util/validation/limits.go (hunk 2 above)
Comment (colega) [reply pkg/util/validation/limits.go]: Did this in some commit below. Thanks.

File: pkg/ingester/lifecycle_test.go (hunk 1 above)
Comment (colega) [reply pkg/ingester/lifecycle_test.go]: After talking on Slack, I changed the behavior to interpret 0 as disabled.

**Comment by colega** [comment]
I'll wait for @jdbaldry's review on the runbooks before merging.

File: docs/sources/mimir/manage/mimir-runbooks/_index.md [hunk 3]
```diff
@@ -1698,6 +1698,31 @@ On a per-tenant basis, you can fine tune the tolerance by configuring the `creat
...
+
+{{< admonition type="note" >}}
+Only the invalid samples are skipped during the ingestion. Valid samples within the same request are still ingested.
+{{< /admonition >}}
+
+### err-mimir-exemplar-too-far-in-past
+
+This non-critical error occurs when Mimir rejects an exemplar because its timestamp is too far in the past compared to the wall clock.
+
+See [`err-mimir-too-far-in-past`](#err-mimir-too-far-in-past) for more details and how to fix it.
```
Comment (jdbaldry) [inline docs/sources/mimir/manage/mimir-runbooks/_index.md:1724]: We prefer "refer to" over "see"

```suggestion
Refer to [`err-mimir-too-far-in-past`](#err-mimir-too-far-in-past) for more details and how to fix it.
```

**Review by jdbaldry — APPROVED** [review]
Docs LGTM and are consistent with the rest of the runbook format. 

nit: consider using contractions to match our tone guidelines: https://grafana.com/docs/writers-toolkit/write/style-guide/voice-tone-guidelines/#tone

File: docs/sources/mimir/manage/mimir-runbooks/_index.md [hunk 4]
```diff
@@ -1698,6 +1698,31 @@ On a per-tenant basis, you can fine tune the tolerance by configuring the `creat
...
 
+### err-mimir-too-far-in-past
+
+This non-critical error occurs when Mimir rejects a sample because its timestamp is too far in the past compared to the wall clock.
+
+How it **works**:
+
+- The distributor or the ingester implements an lower limit on the timestamp of incoming samples, it is used to protect the system from potential abuse or mistakes.
+- The lower limit is defined by the current wall clock minus the `out_of_order_time_window` and minus the `past_grace_period` settings.
+- The samples that are too far in the past are not ingested.
```
Comment (colega) [inline docs/sources/mimir/manage/mimir-runbooks/_index.md:1709]:
```suggestion
- The samples that are too far in the past aren't ingested.
```

**Pull request merged**